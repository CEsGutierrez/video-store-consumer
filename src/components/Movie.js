import React from 'react';
import PropTypes from 'prop-types';

const Movie = ({id, title, overview, release_date, image_url, external_id, selectMovieCallback}) => {
  return (
    <div>
      {title}
      <button
          // className="btn btn-primary pet-card--select-pet-btn"
          onClick={() => { selectMovieCallback(id, 'movieList', 'currentMovie' ) }}
        >
          Select
      </button>
    </div>
  )
}

export default Movie;