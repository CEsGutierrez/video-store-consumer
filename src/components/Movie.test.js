import React from 'react';
import { render, cleanup } from '@testing-library/react'
import Movie from './Movie';

describe('Movie', () => {
  test('that it matches the existing snapshot', () => {
    // Arrange-Act
    const { asFragment } = render(
      <Movie
      id={1}
      title=''
      overview=''
      release_date=''
      image_url=''
      external_id={3}
      selectMovieCallback={() => { }}
      />
    );

    // Assert
    expect(asFragment()).toMatchSnapshot();
    cleanup();
  });

  test('It will render the proper movie fields', () => {
    // Arrange & Act
    const container = render(<Movie
      id={1}
      title='National Treasure: Book of Secrets'
      overview=''
      release_date=''
      image_url=''
      external_id={3}
      selectMovieCallback={() => { }}
      />);

    // Assert
    expect(container.getByText(/National Treasure: Book of Secrets/)).toBeDefined();
  });

  test('The "selectMovieCallback" function is called when the `select` button is clicked on', () => {

    const selectMovie = jest.fn();

    const container = render(<Movie
      id={1}
      title=''
      overview=''
      release_date=''
      image_url=''
      external_id={3}
      selectMovieCallback={selectMovie}
      />);

    const selectButton = container.getByText(/Select/);
    selectButton.click();

    expect(selectMovie).toHaveBeenCalled();
  });

});