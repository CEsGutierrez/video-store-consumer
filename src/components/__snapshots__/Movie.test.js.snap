// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie that it matches the existing snapshot 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <section
      class="card__content"
    >
      <img
        alt="Movie cover art"
        class="card__content-image"
        src=""
      />
      <section
        class="card__content-text"
      />
      <button
        class="card__select"
      >
        Select
      </button>
    </section>
  </div>
</DocumentFragment>
`;
